### PR TITLE
Add framework resolution tests for --fx-version

### PR DESCRIPTION
**Pull request by vitek-karas**

### CODE REVIEW

File: src/test/HostActivationTests/FrameworkResolution/RollForwardOnNoCandidateFxSettings.cs [hunk 1]
```diff
@@ -198,6 +198,74 @@ public void NoInheritance_MoreRestrictive(SettingLocation settingLocation, strin
...
+
+        [Theory]
+        [InlineData(null, null)]
+        [InlineData(0, null)]
+        [InlineData(0, true)]
+        [InlineData(1, null)]
+        [InlineData(1, true)]
+        [InlineData(2, null)]
+        [InlineData(0, false)]
+        public void FxVersionCLIIgnoresOtherSettings(int? rollForwardOnNoCandidateFx, bool? applyPatches)
```
Comment (elinor-fung) [inline src/test/HostActivationTests/FrameworkResolution/RollForwardOnNoCandidateFxSettings.cs]: Maybe make `FxVersionCLI_` the prefix? e.g. `FxVersionCLI_IgnoresOtherSettings`, `FxVersionCLI_AppliesToFirstFrameworkReference_NETCoreAppFirst`
Seems more in line with the naming of the other framework resolution tests.

File: src/test/HostActivationTests/FrameworkResolution/RollForwardOnNoCandidateFxSettings.cs [hunk 2]
```diff
@@ -198,6 +198,74 @@ public void NoInheritance_MoreRestrictive(SettingLocation settingLocation, strin
...
 
+        [Theory]
+        [InlineData("1.0.0", "2.5.5", "2.5.5")]
+        [InlineData("2.0.0", "2.5.5", "2.5.5")]
+        [InlineData("2.5.4", "2.5.5", "2.5.5")]
+        [InlineData("2.5.5", "2.5.5", "2.5.5")]
+        [InlineData("2.5.5", "2.5.4", "2.5.4")]
+        [InlineData("2.5.5", "2.5.3", null)]
+        [InlineData("2.5.5", "2.5.5-preview1", null)]
+        public void FxVersionCLIOverride(string frameworkReferenceVersion, string fxVersion, string resolvedFramework)
```
Comment (elinor-fung) [inline src/test/HostActivationTests/FrameworkResolution/RollForwardOnNoCandidateFxSettings.cs]: Maybe I'm missing something, but I'm not sure how these are related to `RollForwardOnNoCandidateFxSettings`? They seem to just be testing setting fx version?

File: src/test/HostActivationTests/Constants.cs [hunk 3]
```diff
@@ -17,5 +17,11 @@ public static class RollForwardOnNoCandidateFxSetting
             public const string CommandLineArgument = "--roll-forward-on-no-candidate-fx";
             public const string EnvironmentVariable = "DOTNET_ROLL_FORWARD_ON_NO_CANDIDATE_FX";
         }
+
+
```
Comment (elinor-fung) [inline src/test/HostActivationTests/Constants.cs]: Extra newline.

File: src/test/HostActivationTests/FrameworkResolution/RollForwardOnNoCandidateFxSettings.cs (hunk 1 above)
Comment (vitek-karas) [reply src/test/HostActivationTests/FrameworkResolution/RollForwardOnNoCandidateFxSettings.cs]: I moved them into a separate class, so no prefix at all.

File: src/test/HostActivationTests/FrameworkResolution/RollForwardOnNoCandidateFxSettings.cs (hunk 2 above)
Comment (vitek-karas) [reply src/test/HostActivationTests/FrameworkResolution/RollForwardOnNoCandidateFxSettings.cs]: Yeah … I was kind of torn … it made it easy :-)…
I moved them out into a separate class. There is still overlap, but much smaller.

**Pull request merged**